### PR TITLE
Move event processing to separate classes

### PR DESCRIPTION
**Pull request by calina-c**
## Description
Since we want new events implemented, it's important we separate some concerns. Right now, the EventsMonitor does more than it should: monitors events, decrypts assets and handles event processing. I've moved the event processing and decryption functionalities to different classes.

## Is this PR related with an open issue?

Related to Issue #403.

## Types of changes
Enhancement.

@DMats is also helping me test this with a properly setup env, but anyone feel free to review until them.

### CODE REVIEW

File: aquarius/events/processors.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+        logger.debug(f"checking allowed publishers: {publisher_address}")
+        if not self.allowed_publishers:
+            return True
+
+        publisher_address = self._web3.toChecksumAddress(publisher_address)
+        return publisher_address in self.allowed_publishers
+
+    def make_record(self, data):
+        _record = init_new_ddo(data, self.timestamp)
+        # this will be used when updating the doo
```
Comment (DMats) [inline aquarius/events/processors.py:62]: Typo. I believe doo should be ddo. Also, what "this" is this comment referring to?

File: aquarius/events/processors.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+                f"encountered an error while saving the asset data to OceanDB: {str(err)}"
+            )
+            return False
+
+
+class MetadataUpdatedProcessor(EventProcessor):
+    def make_record(self, data, asset):
+        _record = init_new_ddo(data, self.timestamp)
+        # make sure that we do not alter created flag
+        _record["created"] = asset["created"]
```
Comment (DMats) [inline aquarius/events/processors.py:152]: Should this be an assertion instead of assignment?

Comment (DMats) [reply aquarius/events/processors.py:152]: Please disregard my previous comment.  This line needs to be an assignment

**Review by DMats — APPROVED** [review]
I left lots of comments and many of them I ended up taking back :facepalm: The remaining ones should be pretty easy to address. Great PR. I'm going to pre-emptively approve.

**Review by DMats — APPROVED** [review]

**Comment by calina-c** [comment]
This is temporarily abandoned. We need to rethink some aspects of Aquarius, not sure if this is one of them.

**Comment by calina-c** [comment]
To be merged in `stable-aqua` after a review of that branch.

**Pull request merged**